pkg/engine/internal/planner/physical/planner.go

```diff
@@ -229,6 +234,16 @@ func (p *Planner) processMakeTable(lp *logical.MakeTable, ctx *Context) (Node, e
 		}
 	}
 
+	// resolve ambiguous predicates to metadata type where applicable to improve scanning
+	unambiguousPredicates := make([]Expression, 0, len(predicates))
+	for _, predicate := range predicates {
+		p, changed := disambiguateExpression(predicate, metadataColumns)
```

Comment:
I was originally thinking to do this via a negative check: Can we decide if this is definitely not a stream label, and therefore must be metadata or parsed?

The advantage of that way is that there is likely a lot more metadata labels than stream labels, so there are lower data requirements to pass around the stream labels. That said, in either case we only need to find labels actually mentioned in the query which should be a pretty small set to start with.